testdata/p4_16_samples_outputs/action-register-midend.p4

```diff
@@ -0,0 +1,32 @@
+#include <core.p4>
+
+extern Local<D> {
+    Local();
+    D read();
+    void write(in D data);
+}
+
+control C() {
+    @name("C.tmp") bit<32> tmp;
+    @noWarn("unused") @name(".NoAction") action NoAction_1() {
+    }
+    @name("C.RW") action RW() {
+        @name("C.local") Local<bit<32>>() local_0;
```

Comment:
Does this change permit multiple actions in the same control C that each declare a local extern instance with the same name `local`?  Local declarations inside of action bodies definitely seems to imply that this should be possible.

Note that adding the action name to the auto-generated `@name` annotation IS NOT ENOUGH to make that possible right now, because we also allow multiple actions with the same name to be defined in the same control.

Reply:
two externs declarations in separate actions will be distinct objects, even if they have the same local name.

Reply:
The `@name` annotation "C.local" is "enclosing_control_name.local_instance_name_inside_action", yes?

I guess I will try out this PR's changes myself and find out, but what if there was another action that also had an instance with the same name `local` inside of control `C`?  Would their `@name` annotations be identical?  That should be considered a bug, shouldn't it?

Reply:
See the attached test program action-register2.p4, which is just a minor variation of your action-register.p4.  Look at the output program from the last FrontEnd pass.  It has two DirectRegister instances with `@name("C.local")`.
 It seems like a bug to me that p4c is intentionally generating identical `@name` annotations for two different instances.

If you do not consider that a bug, why not?

What can a control plane API generation pass do with this that is possibly useful?

Reply:
Attaching zip file this time.
[p4c-pr-3938-maybe-bug.zip](https://github.com/p4lang/p4c/files/11070682/p4c-pr-3938-maybe-bug.zip)

Reply:
Note: I am not saying that I think if this is considered a bug, that it needs to be fixed right away.  It would be best to worry about it only if we do move forward with enhancing the language with the ability to instantiate extern objects within action bodies.

Reply:
The right solution is to also use the action name as part of the generated name.

Reply:
But don't we allow more than one action with the same name, too?

Reply:
I am not sure we do. Even if we do, they can use annotations to specify different names.